### ISSUE TITLE
Reweight loss for PE-Net-Reweight

### ISSUE THREAD

**Issue by KanghoonYoon** [VL-Group/PENET#7]
Thank you for offering your code.

I would like to implement the PE-Net reweight, but the losses of PE-Net consists of four losses.

Did you use the reweight method for both L_{e_sim} and L_{e_euc}, or only for L_{e_sim} ?

Thank you

**Comment by KanghoonYoon** [comment]
I will appreciate if you explain how to set the beta and the beta value of the reweighting function [1] that you referred.

[1] Class-Balanced Loss Based on Effective Number of Samples

**Comment by aa200647963** [comment]
Thank you for your interest in our work, we only use the class-balanced loss for "Prototype-based Learning  ---- cosine similarity".  Besides, we set the beta of the class-balanced loss to 0.99999.

**Comment by KanghoonYoon** [comment]
@aa200647963 Thank you for the early response. When the loss based on the cosine similarity (L_e_sim) is rescaled by the frequency, the loss may be relatively small compared to the other three losses. 

**Do you just rescale with (1-beta)/(1-beta**{freqeuncy}) ? Or Do you multiply some constant after rescaling the L_{e_sim} with it?.** 

**And, I would like to ask how to decide the weight for __background__ classes.**

Thank you.

**Comment by KanghoonYoon** [comment]
Do you have the code release plan for PE-Net (reweight) ? Although I tried to implement it, using reweighting loss sacrifices a lot of recall (R@100=0.3, mR@100=0.44 under the Predcls task), which is a very different result from your paper.

**Comment by Jeonghoon4** [comment]
Hello,  I have recently become interested in SGG research, and I find your work very intriguing. Thank you for sharing your code. I also have the same question about reproducing the results of PE-Net-Reweight in the paper.

Following the previous comment(https://github.com/VL-Group/PENET/issues/2#issuecomment-1493560685), I used the class weight (self.pred_weight) with cross entropy loss for relation prototype loss.

Below is the validation result at 35,000 iterations (PredCls, VG, current training). Even so, it is similar to the result reported in the paper. However, since I only used the computed class-specific weights, I would like to know if there is any difference compared to the class weights reported in the paper for PE-Net-Reweight.

![Image](https://github.com/user-attachments/assets/5ef84cb5-ab09-4a36-9ea9-3982fa040da7)

**Comment by missyoudaisy** [comment]
> Do you have the code release plan for PE-Net (reweight) ? Although I tried to implement it, using reweighting loss sacrifices a lot of recall (R@100=0.3, mR@100=0.44 under the Predcls task), which is a very different result from your paper.

Hello, do you realize the result of PE-Net-Reweight in the official paper?

**Comment by missyoudaisy** [comment]
> Hello, I have recently become interested in SGG research, and I find your work very intriguing. Thank you for sharing your code. I also have the same question about reproducing the results of PE-Net-Reweight in the paper.
> 
> Following the previous comment([#2 (comment)](https://github.com/VL-Group/PENET/issues/2#issuecomment-1493560685)), I used the class weight (self.pred_weight) with cross entropy loss for relation prototype loss.
> 
> Below is the validation result at 35,000 iterations (PredCls, VG, current training). Even so, it is similar to the result reported in the paper. However, since I only used the computed class-specific weights, I would like to know if there is any difference compared to the class weights reported in the paper for PE-Net-Reweight.
> 
> ![Image](https://github.com/user-attachments/assets/5ef84cb5-ab09-4a36-9ea9-3982fa040da7)

Hello, do you realize the result of PE-Net-Reweight in the official paper?
